nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/queue/GetAzureQueueStorage.java

```diff
@@ -0,0 +1,208 @@
+            .required(true)
+            .defaultValue("30 secs")
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .build();
+
+    private static final List<PropertyDescriptor> properties = Collections.unmodifiableList(Arrays.asList(
+            AzureStorageUtils.ACCOUNT_NAME, AzureStorageUtils.ACCOUNT_KEY, AzureStorageUtils.PROP_SAS_TOKEN, QUEUE, AUTO_DELETE,
+            BATCH_SIZE, VISIBILITY_TIMEOUT));
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return Collections.singleton(REL_SUCCESS);
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+
+        final int visibilityTimeoutInSecs = context.getProperty(VISIBILITY_TIMEOUT).asTimePeriod(TimeUnit.SECONDS).intValue();
+        final int batchSize = context.getProperty(BATCH_SIZE).asInteger();
+        final boolean autoDelete = context.getProperty(AUTO_DELETE).asBoolean();
+
+        final Iterable<CloudQueueMessage> retrievedMessagesIterable;
+
+        try {
+            retrievedMessagesIterable = cloudQueue.retrieveMessages(batchSize, visibilityTimeoutInSecs, null, null);
+        } catch (final StorageException e) {
+            getLogger().error("Failed to retrieve messages from the provided Azure Storage Queue due to {}", new Object[] {e});
+            context.yield();
+            return;
+        }
+
+        final List<CloudQueueMessage> cloudQueueMessages = toList(retrievedMessagesIterable);
+
+        for (final CloudQueueMessage message : cloudQueueMessages) {
+            FlowFile flowFile = session.create();
+
+            final Map<String, String> attributes = new HashMap<>();
+
+            attributes.put("azure.queue.uri", cloudQueue.getUri().toString());
+            attributes.put("azure.queue.insertionTime", message.getInsertionTime().toString());
+            attributes.put("azure.queue.expirationTime", message.getExpirationTime().toString());
+            attributes.put("azure.queue.messageId", message.getMessageId());
+            attributes.put("azure.queue.popReceipt", message.getPopReceipt());
+
+            flowFile = session.putAllAttributes(flowFile, attributes);
+            flowFile = session.write(flowFile, out -> {
+                try {
+                    out.write(message.getMessageContentAsByte());
+                } catch (StorageException e) {
+                    getLogger().error("Failed to write the retrieved queue message to FlowFile content due to {}", new Object[] {e});
+                    context.yield();
+                }
+            });
+
+            session.transfer(flowFile, REL_SUCCESS);
+            session.getProvenanceReporter().receive(flowFile, cloudQueue.getStorageUri().toString());
+        }
+
+        if(autoDelete) {
+            session.commit();
```

Comment:
should we clearly indicate that in case of issues we could create duplicates? (in case we committed the session but didn't delete the messages from the queue)

Reply:
Done